app/uk/gov/hmrc/agentmapping/repository/MappingRepository.scala

```diff
@@ -82,11 +82,10 @@ abstract class BaseMappingRepository[T: Format: Manifest](
         Seq("utr" -> Ascending, identifierKey -> Ascending),
         Some("utrAndIdentifier"),
         unique = true,
-        partialFilter = Some(BSONDocument("utr" -> BSONDocument("$exists" -> true))),
-        options = BSONDocument("expireAfterSeconds" -> 86400)
+        partialFilter = Some(BSONDocument("utr" -> BSONDocument("$exists" -> true)))
       ),
       Index(Seq("arn" -> Ascending), Some("AgentReferenceNumber")),
-      Index(Seq("utr" -> Ascending), Some("Utr"))
+      Index(Seq("utr" -> Ascending), Some("Utr"), options = BSONDocument("expireAfterSeconds" -> 86400))
```

Comment:
I think the TTL index needs to be on a datetime field. e.g.:
```
      Index(
        key = Seq("createdDate" -> IndexType.Ascending),
        name = Some("createDate"),
        unique = false,
        options = BSONDocument("expireAfterSeconds" -> 86400)
      )
```

Reply:
Make sure createdDate is the correct type (DATE) in the db ;) Or it won't work